### PR TITLE
Support/coin hedera

### PR DESCRIPTION
**Pull request by Wozacosta**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

_Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
_For libraries, you can add a code sample of how to use it._
_For bug fixes, you can explain the previous behaviour and how it was fixed._
_In case of visual features, please attach screenshots or video recordings to demonstrate the changes._

### ❓ Context

- **JIRA or GitHub link**: [LIVE-8171](https://ledgerhq.atlassian.net/browse/LIVE-8171)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-8171]: https://ledgerhq.atlassian.net/browse/LIVE-8171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/coin-modules/coin-hedera/package.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    },
+    "./package.json": "./package.json"
+  },
+  "license": "Apache-2.0",
+  "dependencies": {
+    "@ledgerhq/coin-framework": "workspace:^",
+    "@ledgerhq/cryptoassets": "workspace:^",
+    "@ledgerhq/devices": "workspace:^",
+    "@ledgerhq/errors": "workspace:^",
+    "@ledgerhq/live-config": "workspace:^",
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-hedera/package.json:58]: This dependency shouldn't exist in a coin-module.
The CoinConfiguration must be independent from LiveConfiguration.

File: libs/coin-modules/coin-hedera/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ConfigInfo } from "@ledgerhq/live-config/LiveConfig";
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-hedera/src/config.ts:1]: You can see other coin-module, but the type of the config of a coin-module, must not depend on LiveConfig package.

**Review by sprohaszka-ledger — CHANGES_REQUESTED** [review]
Quickly I see on dependency that should not exists.

File: libs/coin-modules/coin-hedera/src/config.ts (hunk 2 above)
Comment (qperrot) [reply libs/coin-modules/coin-hedera/src/config.ts:1]: @sprohaszka-ledger I follow what the guys did for `cosmos` and we do have the same import here https://github.com/LedgerHQ/ledger-live/blob/77846a0be4c2a08dfef0e938e1af1479ad3787b7/libs/coin-modules/coin-cosmos/src/config.ts#L3

I will do the modification on `hedera` but should we open a ticket for `cosmos` ?

Comment (Wozacosta) [reply libs/coin-modules/coin-hedera/src/config.ts]: I'll modify `cosmos` and open up a PR

File: libs/coin-framework/src/bridge/jsHelpers.ts [hunk 3]
```diff
@@ -545,8 +546,21 @@ export function makeAccountBridgeReceive<A extends Account = Account>(
       getAddressFn(deviceId, arg).then(r => {
         const accountAddress = account.freshAddress;
 
-        if (verify && r.address !== accountAddress) {
-          throw new WrongDeviceForAccount();
+        if (verify) {
+          // NOTE: could pass a param to make different checks for different currencies
+          if (account.currency.id === "hedera") {
```
Comment (hedi-edelbloute) [inline libs/coin-framework/src/bridge/jsHelpers.ts:551]: We do not want this kind of check in common code, is there another way to do this ?

Comment (Wozacosta) [reply libs/coin-framework/src/bridge/jsHelpers.ts:551]: Good point, maybe by passing additional params to make the verify happen on the provided vars instead, wdyt?. 
For now we could start with a boolean `verifyUsingPublicKey`, as hedera seems to be the sole chain to do a different so. 

So something like 
```js
if (verify) {
          // NOTE: could pass a param to make different checks for different currencies
          if (verifyUsingPublicKey && r.publicKey !== account.seedIdentifier) {
```

File: libs/coin-framework/src/bridge/jsHelpers.ts [hunk 4]
```diff
@@ -517,6 +517,7 @@ export function makeAccountBridgeReceive<A extends Account = Account>(
   getAddressFn: GetAddressFn,
   {
     injectGetAddressParams,
+    // NOTE: might pass params here
```
Comment (hedi-edelbloute) [inline libs/coin-framework/src/bridge/jsHelpers.ts:520]: We can delete this comment

**Review by hedi-edelbloute — DISMISSED** [review]

**Review by hedi-edelbloute — DISMISSED** [review]

**Review by hedi-edelbloute — DISMISSED** [review]

File: libs/coin-modules/coin-hedera/src/types/index.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export * from "./bridge";
+export * from "./signer";
+export * from "../errors";
```
Comment (Wozacosta) [inline libs/coin-modules/coin-hedera/src/types/index.ts:3]: can remove the export from errors here

**Review by themooneer — DISMISSED** [review]
LGTM

**Review by KVNLS — DISMISSED** [review]

**Review by cgrellard-ledger — DISMISSED** [review]

**Review by KVNLS — APPROVED** [review]

File: libs/coin-modules/coin-hedera/src/bridge/index.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  };
+}
+
+export function createBridges(signerContext: SignerContext<HederaSigner>) {
+  return {
+    currencyBridge: buildCurrencyBridge(signerContext),
+    accountBridge: buildAccountBridge(signerContext),
+  };
+}
+export { prepareTransaction, estimateMaxSpendable };
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-hedera/src/bridge/index.ts:61]: We don't want to expose those methods outside of the bridge.

File: libs/coin-modules/coin-hedera/src/bridge/signOperation.ts [hunk 7]
```diff
@@ -27,9 +25,12 @@ export const signOperation: AccountBridge<Transaction>["signOperation"] = ({
...
 
-          await hederaTransaction.signWith(accountPublicKey, async bodyBytes => {
-            return await new Hedera(transport).signTransaction(bodyBytes);
-          });
+          const res = (await signerContext(deviceId, async signer => {
+            await hederaTransaction.signWith(accountPublicKey, async bodyBytes => {
+              return await signer.signTransaction(bodyBytes);
+            });
+            return hederaTransaction.toBytes();
+          })) as HederaSignatureSdk;
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-hedera/src/bridge/signOperation.ts:33]: Not sure to understand why you force cast here.

**Review by sprohaszka-ledger — CHANGES_REQUESTED** [review]
Small changes 👍

**Review by sprohaszka-ledger — DISMISSED** [review]
LGTM

**Review by KVNLS — APPROVED** [review]

**Review by themooneer — APPROVED** [review]

**Review by hedi-edelbloute — APPROVED** [review]

**Pull request merged**